http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/LeoAuthProviderHelper.scala

```diff
@@ -4,19 +4,22 @@ import akka.actor.ActorSystem
 import akka.http.scaladsl.model.StatusCodes
 import com.typesafe.config.Config
 import com.typesafe.scalalogging.LazyLogging
+import java.util.UUID.randomUUID
 import net.ceedubs.ficus.Ficus._
 import org.broadinstitute.dsde.workbench.leonardo.model._
 import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterName
-import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail}
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo, WorkbenchEmail}
 import org.broadinstitute.dsde.workbench.model.google.GoogleProject
 import org.broadinstitute.dsde.workbench.util.FutureSupport
 
 import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future, TimeoutException}
 import scala.util.control.NonFatal
 
-case class AuthProviderException(authProviderClassName: String, isTimeout: Boolean = false)
-  extends LeoException(s"Call to $authProviderClassName auth provider ${if (isTimeout) "timed out" else "failed"}", StatusCodes.InternalServerError)
+case class AuthProviderException(authProviderClassName: String,
+                                 isTimeout: Boolean = false,
+                                 traceId: TraceId)
+  extends LeoException(s"[$traceId] Call to $authProviderClassName auth provider ${if (isTimeout) "timed out" else "failed"}", StatusCodes.InternalServerError)
```

Comment:
Do we want to make the status code in `AuthProviderException` match the status code from Sam (instead of `InternalServerError`)?

Reply:
@Qi77Qi and I had talked faces about this and we thought it's still appropriate to return `500` because the error is not due to the user's (in the perf test case, cluster creation) request and there isn't an action they can take. If anything, it feels that the error message isn't helpful to the user and I wonder if we should change it to something more high level.

Reply:
I think the other reason is swagger generated client doesn't really give you the status code from sam other than throwing an exception. (If this is not true, it might still be useful to give user 404 etc if it's 404 from sam)

Reply:
Yea, I think there's that too.

Reply:
OK, that makes sense.

That's a good point about the error message - could we change it to something more user friendly than `SamAuthProvider failed`?